plasmapy/particles/particle_collections.py

```diff
@@ -0,0 +1,207 @@
+    >>> particle_list[0]
+    Particle("e-")
+    >>> particle_list.mass
+    <Quantity [9.1093...e-31, 9.1093...e-31] kg>
+    >>> particle_list.charge
+    <Quantity [-1.60217663e-19,  1.60217663e-19] C>
+    >>> particle_list.symbols
+    ['e-', 'e+']
+    >>> particle_list + [Particle("p+")]
+    ParticleList(['e-', 'e+', 'p+'])
+    """
+
+    @staticmethod
+    def _list_of_particles_and_custom_particles(
+        particles: Iterable[ParticleLike],
+    ) -> List[Union[Particle, CustomParticle]]:
+        """
+        Convert an iterable that provides `ParticleLike` objects into a
+        `list` containing `Particle` and `CustomParticle` instances.
+        """
+        new_particles = []
+        for obj in particles:
+            if isinstance(obj, (Particle, CustomParticle)):
+                new_particles.append(obj)
+            elif isinstance(obj, DimensionlessParticle):
+                raise TypeError(
+                    "ParticleList instances cannot include dimensionless particles."
+                )
+            else:
+                try:
+                    new_particles.append(Particle(obj))
+                except (TypeError, InvalidParticleError) as exc:
+                    raise InvalidParticleError(
+                        f"The object {obj} supplied to ParticleList is not a "
+                        f"particle-like object."
+                    ) from exc
+
+        return new_particles
+
+    def __init__(self, particles: Iterable):
+        self._data = self._list_of_particles_and_custom_particles(particles)
+
+    @staticmethod
+    def _cast_other_as_particle_list(other):
+        if isinstance(other, ParticleList):
+            return other
+
+        try:
+            return ParticleList(other)
+        except (InvalidParticleError, TypeError):
+            pass
+
+        try:
+            return ParticleList([other])
+        except (InvalidParticleError, TypeError):
+            raise InvalidParticleError(f"Cannot cast {other} into a ParticleList")
+
+    def __add__(self, other):
+        try:
+            other_as_particle_list = self._cast_other_as_particle_list(other)
+        except (TypeError, InvalidParticleError) as exc:
+            raise InvalidParticleError(
+                f"Cannot add {repr(other)} to a ParticleList."
+            ) from exc
+        return ParticleList(self.data + other_as_particle_list.data)
+
+    def __radd__(self, other):
+        other_as_particle_list = self._cast_other_as_particle_list(other)
+        return other_as_particle_list.__add__(self)
+
+    def __repr__(self):
+        return f"ParticleList({repr(self.symbols)})"
+
+    def __gt__(self, other):
+        from plasmapy.particles.nuclear import nuclear_reaction_energy
+
+        other_as_particle_list = self._cast_other_as_particle_list(other)
+        return nuclear_reaction_energy(
+            reactants=self.symbols, products=other_as_particle_list.symbols
+        )
```

Comment:
This should be documented! The comment on this PR you made earlier would be a great start.